src/components/ProductPage/ProductPage.jsx

```diff
@@ -1,21 +1,41 @@
 import React, { useEffect } from 'react';
-import { useParams, useHistory  } from 'react-router-dom';
+import { useParams, useHistory } from 'react-router-dom';
 import { useDispatch, useSelector } from "react-redux";
 import 'antd/dist/antd.css';
 import { Card as CardAntd } from 'antd';
 import 'antd/dist/antd.css';
-import { productPageSelectors } from 'store/productPageSlice';
-import { fetchProduct } from 'store/productPageSlice';
+import { productPageSelectors, fetchProduct } from 'store/productPageSlice';
+import { cartSelectors, cartActions } from 'store/cartSlice';
 import css from "./product.module.css"
 import { Loader } from "../common"
-
+import { Header } from 'components/Header';
 
 export const ProductPage = () => {
+
     const { id } = useParams()
 
     const history = useHistory();
 
-    const products = useSelector(productPageSelectors.getProduct)
+    const product = useSelector(productPageSelectors.getProduct)
+
+    const good = useSelector(cartSelectors.getGoodById(id));
+
+    const count = good ? good.count : 0
```

Comment:
```suggestion
    const count = good?.count ?? 0
```
? - будет читать свойство только если объект существует
?? - проверяет левую часть от себя и если там null/undefined возвращает правую, иначе левую